compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java

```diff
@@ -435,6 +435,8 @@ static void compileInternal(ModuleContext moduleContext, CompilerContext compile
             packageCache.putSymbol(pkgNode.packageID, pkgNode.symbol);
             compilerPhaseRunner.performTypeCheckPhases(pkgNode);
         } catch (Throwable t) {
+            assert false : "Compilation failed due to" +
```

Comment:
How is -ea is enabled?
Also, can there be any case where the diagnostics will be checked after an assertion is thrown?

Reply:
AFAIK, with TestNG java assertions are enabled by default whereas in the production code assertions are disabled by default.  
You can try running this,
```java
    @Test
    public void testJavaAssert() {
        assert false;
    }
```
If AssertionError is thrown, no way that lines after that are executed.

Reply:
No my question was whether "we"; in our context have any scenario that needed diagnostics to be checked after a unhandled exception

Reply:
Resolving as we discussed this offline.